install/src/command.rs

```diff
@@ -859,14 +864,41 @@ fn semver_of(string: &str) -> Result<semver::Version, String> {
 }
 
 fn check_for_newer_github_release(
-    version_filter: Option<semver::VersionReq>,
+    current_release_semver: &str,
+    semver_update_type: SemverUpdateType,
     prerelease_allowed: bool,
-) -> reqwest::Result<Option<String>> {
-    let mut page = 1;
-    const PER_PAGE: usize = 100;
+) -> Result<Option<String>, String> {
     let client = reqwest::blocking::Client::builder()
         .user_agent("solana-install")
-        .build()?;
+        .build()
+        .map_err(|err| err.to_string())?;
+
+    // If we have a hardcoded version, and we don't care if it's a prerelease, we don't need to stress the API
+    if semver_update_type == SemverUpdateType::Fixed && prerelease_allowed {
```

Comment:
The comment and the code are out of sync here. If we're using a fixed semver, you're right, we don't care if it's a pre-release! This way we can immediately pick up that pre-release if it's what we specified.
```suggestion
    if semver_update_type == SemverUpdateType::Fixed {
```

Reply:
hmm i did it this way because `prerelease_allowed` in `check_for_newer_github_release` is a synonym for `is_init` in its caller, `init_or_update`. so in an update context, it needs to be able to check if a given version is a prerelease. but... now im looking at the interface itself and it seems like its actually not possible to pass a version to update, so i think the code path i envisioned covering doesnt actually exist

Reply:
It's a confusing interface, but I think `SemverUpdateType::Fixed` already encapsulates that in this call, right? Which means that this is perfectly correct, just redundant.